### DISCUSSION TITLE
Application orchestrator dependency check had an unexpected error on MacOS

### DISCUSSION THREAD

**Discussion by julionav** [microsoft/aspire#3047] [Q&A] [answered]
## Problem

When running Aspire Preview 4 on MacOS, I get the following error:
```
Application orchestrator dependency check had an unexpected error System.InvalidOperationException: Command /usr/local/share/dotnet/packs/Aspire.Hosting.Orchestration.osx-arm64/8.0.0-preview.3.24105.21/tools/dcp info returned non-zero exit code 1
   at Aspire.Hosting.Dcp.DcpDependencyCheck.EnsureDcpDependenciesAsync(CancellationToken cancellationToken) in /_/src/Aspire.Hosting/Dcp/DcpDependencyCheck.cs:line 67.
```

It seems that the preview-3 orchestrator is being used instead of version preview-4?

**Comment by davidfowl** [comment]
What do you see when you run `dotnet workload list`?

**Comment by julionav** [comment]
> What do you see when you run `dotnet workload list`?

I see:
![CleanShot 2024-03-20 at 11 41 12@2x](https://github.com/dotnet/aspire/assets/45607850/cbe80996-456e-4564-b4f1-e37b548eb284)

My installed packages:
![CleanShot 2024-03-20 at 11 41 43@2x](https://github.com/dotnet/aspire/assets/45607850/beffc4f5-c1c0-40b5-bcb1-6827028f82b4)

**Comment by julionav** [comment]
I added Aspire using Rider, which added preview.3, then updated the nuget package. Maybe thats the issue?

**Comment by julionav** [comment, marked as answer]
Running `sudo dotnet workload install aspire` fixed the issue! Thanks for pointing me on the right direction @davidfowl
